sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -185,6 +187,25 @@ public static BinaryData fromBytes(byte[] data) {
         return new BinaryData(data);
     }
 
+    /**
+     * Serialize the given {@link Object} into {@link BinaryData} using the provided {@link JsonSerializer}. This will
+     * require the client to configure Json serializer in classpath.
+     *
+     * @param data The {@link Object} which needs to be serialized into bytes.
+     * @throws NullPointerException if {@code data} is null.
+     * @throws IllegalStateException If cannot find any JSON serializer provider on the classpath.
+     * @return {@link BinaryData} representing binary data. Or {@code null} if it fails to serialize the data.
+     *
+     * @see JsonSerializer
+     */
+    public static BinaryData fromObject(Object data) {
+        Objects.requireNonNull(data, "'data' cannot be null.");
+
+        final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        JsonSerializerProviders.createInstance().serialize(outputStream, data);
```

Comment:
@JonathanGiles, @srnagar, I'd like to see us create a guideline around the default serializer to be used when one isn't passed. In some locations we use `JsonSerializer`/`ObjectSerializer` and in others we default to using `JacksonAdapter`. I have a feeling we may need to go down the route of always falling back to `JacksonAdapter`, except with serialization formats we didn't previously support such as Avro and Protobuf, as using `JsonSerializerProviders.createInstance` may result in customer applications throwing an exception when they previously didn't. This restriction may only apply to logic in `azure-core` which could accept an `ObjectSerializer`/`JsonSerializer` as it cannot take a dependency on the implementation libraries.

Reply:
Possible guidance:

If `azure-core` or retrofitting a library, either we cannot accept a null serializer type or we default to `JacksonAdapter`. Reasoning is that we may introduce exceptions into a customer's application that weren't previously being thrown.

If a new library, leverage `JsonSerializerProviders.createInstance()`, and other serializer providers that come in the future, as the default and include a dependency for the default implementation. Reasoning is that it is a new library and should adhere to the best practices for allowing customer configuration of serialization.

Reply:
Set up a meeting for next week and we can discuss this and work on some guidelines text.